src/python/pants/backend/awslambda/python/lambdex.py

```diff
@@ -9,5 +9,5 @@ class Lambdex(PythonToolBase):
     default_version = "lambdex==0.1.3"
     # TODO(John Sirois): Remove when we can upgrade to a version of lambdex with
     # https://github.com/wickman/lambdex/issues/6 fixed.
-    default_extra_requirements = ["setuptools==44.0.0"]
+    default_extra_requirements = ["setuptools==47.3.1"]
```

Comment:
This will break `setup-py` for Python 2 users. I think that this will be okay because we've already had Pytest default to a version that doesn't work with Python 2. So long as we add a warning to https://pants.readme.io/docs/python-setup-py-goal about what to do for Py2 support.

cc @benjyw @stuhood for an ack.

Reply:
Is it worth saying that 2.0 is python 3 only? If it's easy enough to just wait to do this for a few months, that would be preferable...

Reply:
There's no reason to be Python 3 only. It's not very expensive for us to still support Python 2. And we're already in several places defaulting to Python 3-only, but giving you the option to use Python 3. 

Here, I only want to make sure we're aware of the implication of this change and on the same page.

Reply:
I agree with @Eric-Arellano if we it easy/cheap to support python 2, we should.
Even though it is EOL, I think there a people still using it...

Reply:
People working with python2 already have to jump thru hoops, this is just one more. Which I think is fair all things considered.

Reply:
Okay, sounds good. We'll need to update the 2.0 docs then with a warning about Py2 support.

Thanks folks!